src/SubscriptionFactory.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Minishlink\WebPush;
+
+use InvalidArgumentException;
+
+class SubscriptionFactory
+{
+    public static function create(array $array): Subscription
```

Comment:
Instead of an array, parameters should be passed.
This means you will have 2 `create*` methods to cover the cases:

*  One with parameters `$endpoint`, `$p256dh`, `$auth` and `$encoding`
* Another with `$endpoint`, `$public_key`, `$auth_token` and `$encoding`

Reply:
As long as this is ok, I'm fine with that. I was trying to maintain compatibility with most of the example use cases I had seen in the README where these payloads are fetched from a persistence layer or something as arrays and chunked through to build up Subscription objects. Would we want to provide for the DX of supporting that, or safe to just drop it?